### ISSUE TITLE
[2.3] `withTiming` never updates `backgroundColor` on iOS

### ISSUE THREAD

**Issue by nandorojo** [software-mansion/react-native-reanimated#2751]
## Description

If you set `backgroundColor` using `withTiming`, it never changes from its initial value. It worked with 2.2, but not with 2.3.

It works fine on Web, but breaks on iOS.

### Expected behavior

This should work:

```ts
const pressed = useSharedValue(false)
const style = useAnimatedStyle(() => {
  return {
    backgroundColor: withTiming(pressed.value ? '#ffffff' : '#000000')
  }
})
```

### Actual behavior & steps to reproduce

The color never updates if you wrap it with `withTiming`. If you remove `withTiming`, then it does indeed update. 

Here is a video showing what happens:

https://www.loom.com/share/0b78eee16d644a4d8993d7844b328c00

## Snack or minimal code example

<details>
<summary>View Code</summary>

```tsx
import React from 'react'
import { Pressable, StyleSheet } from 'react-native'
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated'

export default function HelloWorld() {
  const pressed = useSharedValue(false)
  const style = useAnimatedStyle(() => {
    return {
      backgroundColor: withTiming(pressed.value ? '#ffffff' : '#000000'),
    }
  })
  return (
    <Pressable
      onPressIn={() => (pressed.value = true)}
      style={styles.container}
      onPressOut={() => (pressed.value = false)}
    >
      <Animated.View style={[styles.shape, style]} />
    </Pressable>
  )
}

const styles = StyleSheet.create({
  shape: {
    justifyContent: 'center',
    height: 250,
    width: 250,
    borderRadius: 25,
    marginRight: 10,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    backgroundColor: '#9c1aff',
  },
})
```

</details>

## Package versions

- React Native: 0.63.4
- React Native Reanimated: 2.3.1
- NodeJS:
- Xcode:
- Java & Gradle:

## Affected platforms

- [ ] Android
- [x] iOS
- [ ] Web

**Comment by lucasmotta** [comment]
I am experiencing the exact same issue, but for Android instead. All my background colors are transparent when using `withTiming` - if I remove it, I get the colors as expected.

**Comment by piaskowyk** [comment]
It will be fixed with https://github.com/software-mansion/react-native-reanimated/pull/2776

**Comment by nandorojo** [comment]
Thanks @piaskowyk!

**Comment by QuintonC** [comment]
> It will be fixed with #2776

looks like there is a change requested on your PR. Any updates on when this might be fixed/resolved?
